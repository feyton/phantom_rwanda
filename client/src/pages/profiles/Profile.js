import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { toast } from 'react-toastify';
import camera from '../../assets/camera-solid.svg';
import { ButtonA as Button } from '../../components/Button.js';
import Input from '../../components/Input.js';
import { loginUser } from '../../redux/reducers/authReducer.js';
import { editInfo, saveInfo } from '../../redux/reducers/profileReducer.js';
import { axiosBase } from '../../utils/Api.js';

const Profile = ({
  firstName,
  lastName,
  company,
  mobileNumber,
  address,
  position,
  email,
  image,
  role
}) => {
  const [src, setSrc] = useState(image);
  const [loading, setloading] = useState(false);
  const dispatch = useDispatch();
  const {
    disabled,
    styles,
    hideSave,
    upload,
    editIcon,
    editPicture,
    hideEdit
  } = useSelector((state) => state?.profileInfo);

  const onChange = async (e) => {
    const reader = new FileReader();
    const file = e.target.files[0];
    const data = new FormData();
    data.append('image', file);
    try {
      setloading(true);
      const res = await axiosBase.post('/accounts/upload', data, {
        headers: {
          'Content-Type': 'multipart/form-data'
        }
      });
      const auth = JSON.parse(localStorage.getItem('auth'));
      reader.onloadend = () => {
        setSrc(reader.result);
        auth.user.image = reader.result;
        dispatch(loginUser(auth.user));
      };
      reader.readAsDataURL(file);
      dispatch(saveInfo());
    } catch (error) {
      toast(error?.response?.data?.data?.message || 'Server error');
    } finally {
      setloading(false);
    }
  };

  const onSubmit = async (e) => {
    e.preventDefault();
    const formData = new FormData(e.target);
    const object = {};
    // eslint-disable-next-line no-return-assign
    formData.forEach((value, key) => {
      if (value !== e.target[key].defaultValue) {
        object[key] = value;
      }
    });
    if (Object.keys(object).length !== 0) {
      try {
        setloading(true);
        await axiosBase.post('/accounts/profile', object);
        dispatch(saveInfo());
        const auth = JSON.parse(localStorage.getItem('auth'));
        auth.user.first_name = object?.firstName || firstName;
        dispatch(loginUser(auth.user));
      } catch (error) {
        toast(error?.response?.data?.data?.message);
      } finally {
        setloading(false);
      }
    }
  };

  return (
    <div className="bg-white rounded  w-full shadow-md min-h-[70vh] mt-6 pb-10">
      <div>
        <div className="flex flex-col tex-center md:flex-row lg:flex-row xl:flex-row items-center justify-center bg-gray-100 mb-10 h-80 md:h-60 lg:h-60 xl:h-60 pb-10 font-raleway">
          <span className="w-20 h-20 my-8 md:w-24 lg:w-24 xl:w-24 md:h-24 lg:h-24 xl:h-24 mr-6">
            <div className="relative">
              <label htmlFor={upload}>
                <img
                  src={src}
                  alt="profile-img"
                  id="profile-img"
                  className={`${editPicture} ${
                    loading ? ' opacity-50' : 'opacity-100'
                  } rounded-full object-cover w-20 h-20 md:w-24 lg:w-24 xl:w-24 md:h-24 lg:h-24 xl:h-24`}
                />
                <div
                  className={`${editIcon} absolute bg-gray-100 rounded-full w-5 h-5 left-7 md:left-10 lg:left-10 xl:left-20 top-7 md:top-8 lg:top-16 xl:top-18 cursor-pointer`}
                >
                  {loading ? (
                    <div className="loader" />
                  ) : (
                    <img src={camera} alt="" className="w-3 h-3 m-1" />
                  )}
                </div>
              </label>
            </div>
            <input
              id="photo-upload"
              type="file"
              accept=".jpg, .jpeg, .png"
              onChange={onChange}
            />
          </span>
          <div className="h-20 flex flex-col items-center md:items-baseline lg:items-baseline xl:items-baseline">
            <h1 className="font-black text-lg md:text-2xl lg:text-3xl xl:text-3xl ">
              {firstName} {lastName}
            </h1>
            <h2 className="font-bold text-sm md:text-base lg:text-base xl:text-base ">
              <b className="capitalize text-primary">{position}</b> {company && " at "} {company}
            </h2>
            <h3>{email}</h3>
            <div className="">
              <Button
                name={loading ? 'Sending...' : 'Edit profile'}
                id="edit"
                styles={`${hideEdit} text-white text-sm md:text-base lg:text-base xl:text-base mt-4 px-4 md:px-6 lg:px-6 xl:px-6`}
                onClick={() => dispatch(editInfo())}
              />
            </div>
          </div>
        </div>
      </div>
      <div className="flex flex-col items-center">
        <form action="" onSubmit={(e) => onSubmit(e)}>
          <section className="flex flex-col items-center lg:flex-row xl:flex-row">
            <div className="ml-8">
              <div className="mb-4 block font-raleway">
                <Input
                  htmlFor="FirstName"
                  labelName="First Name"
                  name="firstName"
                  styles={styles}
                  defaultValue={firstName}
                  id="firstName"
                  disabled={disabled}
                />
              </div>
              <div className="mb-4 block font-raleway">
                <Input
                  htmlFor="LastName"
                  labelName="Last name "
                  type="text"
                  name="lastName"
                  styles={styles}
                  defaultValue={lastName}
                  id="lastName"
                  disabled={disabled}
                />
              </div>
            </div>

            {role === 'admin' || role === 'user' ? (
              ''
            ) : (
              <div className="pl-10 lg:pl-24 xl:pl-24">
                <div className="mb-4 block font-raleway">
                  <Input
                    htmlFor="mobileNumber"
                    labelName="Mobile Number "
                    type="text"
                    name="mobileNumber"
                    styles={styles}
                    defaultValue={mobileNumber}
                    id="mobileNumber"
                    disabled={disabled}
                    required
                    pattern="^07[0-9]{8}$"
                  />
                </div>

                <div className="mb-4 block font-raleway">
                  <Input
                    htmlFor="address"
                    labelName="Address"
                    type="text"
                    name="address"
                    styles={styles}
                    defaultValue={address}
                    id="address"
                    disabled={disabled}
                  />
                </div>
              </div>
            )}
          </section>
          <div className="flex">
            <Button
              name="cancel"
              id="cancel"
              type="button"
              styles={`${hideSave} bg-red text-white px-10 ml-8 mt-16`}
            />

            <Button
              name={loading ? 'Sending...' : 'Save'}
              id="save"
              type="submit"
              styles={`${hideSave} text-white px-10 ml-8 mt-16`}
            />
          </div>
        </form>
      </div>
    </div>
  );
};

export default Profile;
