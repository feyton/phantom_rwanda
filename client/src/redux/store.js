import { configureStore } from '@reduxjs/toolkit';
import { authReducer } from './reducers/authReducer.js';
import { busMovementReducer } from './reducers/busMovementReducer.js';
import { counterReducer } from './reducers/counterReducer.js';
import { dashboardReducer } from './reducers/dashboardReducer.js';
import { managementReducer } from './reducers/managementReducer.js';
import { profileReducer } from './reducers/profileReducer.js';
import { simulateReducer } from './reducers/simulateReducer.js';
import { socketReducer } from './reducers/socketReducer.js';
import { updateProfileReducer } from './reducers/updateProfileReducer.js';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    profileUpdates: updateProfileReducer,
    profileInfo: profileReducer,
    auth: authReducer,
    management: managementReducer,
    dashboard: dashboardReducer,
    socket: socketReducer,
    simulate: simulateReducer,
    movement: busMovementReducer
  }
});
