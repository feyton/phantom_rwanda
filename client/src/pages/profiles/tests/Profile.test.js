import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import * as ReactRedux from 'react-redux';
import { mount } from 'enzyme';
import { fireEvent, render, screen } from '@testing-library/react';
import { store } from '../../../redux/store.js';
import Profile from '../Profile.js';

describe('Profile', () => {
  it('should render the Profile component', () => {
    const elem = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <Profile />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
  it('It should test the setOpen function', () => {
    const onChange = jest.fn();
    const component = mount(
      <Provider store={store}>
        <MemoryRouter>
          <Profile onChange={onChange()} />
        </MemoryRouter>
      </Provider>
    );
    const button = component.find('#photo-upload');
    button.simulate('click');
    expect(onChange).toBeCalledTimes(1);
  });
  it('It should test the dispatch function', () => {
    const elem = render(
      <Provider store={store}>
        <MemoryRouter>
          <Profile />
        </MemoryRouter>
      </Provider>
    );

    fireEvent.click(screen.getByTestId('photo-upload'));
    expect(elem).toMatchSnapshot();
  });
});

describe('Profile page functionality tests', () => {
  const onChangeMock = jest
    .spyOn(FileReader.prototype, 'readAsDataURL')
    .mockImplementation(() => null);
  const event = {
    target: {
      files: [
        {
          name: 'image.png',
          size: 50000,
          type: 'image/png'
        }
      ]
    }
  };

  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter>
        <Profile onChange={onChangeMock()} />)
      </MemoryRouter>
    </Provider>
  );

  const uploadBtn = wrapper.find('#photo-upload');
  it('calls handleSubmit function on form submit', () => {
    act(() => {
      uploadBtn.simulate('change', event);
      expect(onChangeMock).toBeCalledTimes(1);
    });
  });
});
