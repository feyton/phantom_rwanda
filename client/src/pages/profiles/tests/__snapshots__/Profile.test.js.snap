// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile It should test the dispatch function 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-white rounded  w-full shadow-md min-h-[70vh] mt-6 pb-10"
      >
        <div>
          <div
            class="flex flex-col tex-center md:flex-row lg:flex-row xl:flex-row items-center justify-center bg-gray-100 mb-10 h-80 md:h-60 lg:h-60 xl:h-60 pb-10 font-raleway"
          >
            <span
              class="w-20 h-20 my-8 md:w-24 lg:w-24 xl:w-24 md:h-24 lg:h-24 xl:h-24 mr-6"
            >
              <div
                class="relative"
              >
                <label
                  for=""
                >
                  <img
                    alt="profile-img"
                    class=" opacity-100 rounded-full object-cover w-20 h-20 md:w-24 lg:w-24 xl:w-24 md:h-24 lg:h-24 xl:h-24"
                    id="profile-img"
                  />
                  <div
                    class="hidden absolute bg-gray-100 rounded-full w-5 h-5 left-7 md:left-10 lg:left-10 xl:left-20 top-7 md:top-8 lg:top-16 xl:top-18 cursor-pointer"
                  >
                    <img
                      alt=""
                      class="w-3 h-3 m-1"
                      src="camera-solid.svg"
                    />
                  </div>
                </label>
              </div>
              <input
                accept=".jpg, .jpeg, .png"
                id="photo-upload"
                type="file"
              />
            </span>
            <div
              class="h-20 flex flex-col items-center md:items-baseline lg:items-baseline xl:items-baseline"
            >
              <h1
                class="font-black text-lg md:text-2xl lg:text-3xl xl:text-3xl "
              >
                 
              </h1>
              <h2
                class="font-bold text-sm md:text-base lg:text-base xl:text-base "
              >
                <b
                  class="capitalize text-primary"
                />
                 
                 
              </h2>
              <h3 />
              <div
                class=""
              >
                <div>
                  <button
                    class="flex text-white text-sm md:text-base lg:text-base xl:text-base mt-4 px-4 md:px-6 lg:px-6 xl:px-6 bg-blue-500 hover:bg-blue-600 text-white font-bold py-2 px-4 rounded-xl m-2 "
                  >
                    Edit profile
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex flex-col items-center"
        >
          <form
            action=""
          >
            <section
              class="flex flex-col items-center lg:flex-row xl:flex-row"
            >
              <div
                class="ml-8"
              >
                <div
                  class="mb-4 block font-raleway"
                >
                  <div
                    class="undefined mb-2 block font-raleway"
                  >
                    <label
                      class="block text-grey-darker text-sm font-bold font-rale mb-2"
                      for="FirstName"
                    >
                      First Name
                    </label>
                    <input
                      class="font-rale border-none rounded w-full py-2 px-3 text-grey-darker outline-hidden bg-white text-md appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                      disabled=""
                      id="firstName"
                      name="firstName"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="mb-4 block font-raleway"
                >
                  <div
                    class="undefined mb-2 block font-raleway"
                  >
                    <label
                      class="block text-grey-darker text-sm font-bold font-rale mb-2"
                      for="LastName"
                    >
                      Last name 
                    </label>
                    <input
                      class="font-rale border-none rounded w-full py-2 px-3 text-grey-darker outline-hidden bg-white text-md appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                      disabled=""
                      id="lastName"
                      name="lastName"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="pl-10 lg:pl-24 xl:pl-24"
              >
                <div
                  class="mb-4 block font-raleway"
                >
                  <div
                    class="undefined mb-2 block font-raleway"
                  >
                    <label
                      class="block text-grey-darker text-sm font-bold font-rale mb-2"
                      for="mobileNumber"
                    >
                      Mobile Number 
                    </label>
                    <input
                      class="font-rale border-none rounded w-full py-2 px-3 text-grey-darker outline-hidden bg-white text-md appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                      disabled=""
                      id="mobileNumber"
                      name="mobileNumber"
                      pattern="^07[0-9]{8}$"
                      required=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="mb-4 block font-raleway"
                >
                  <div
                    class="undefined mb-2 block font-raleway"
                  >
                    <label
                      class="block text-grey-darker text-sm font-bold font-rale mb-2"
                      for="address"
                    >
                      Address
                    </label>
                    <input
                      class="font-rale border-none rounded w-full py-2 px-3 text-grey-darker outline-hidden bg-white text-md appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                      disabled=""
                      id="address"
                      name="address"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </section>
            <div
              class="flex"
            >
              <div>
                <button
                  class="hidden bg-red text-white px-10 ml-8 mt-16 bg-blue-500 hover:bg-blue-600 text-white font-bold py-2 px-4 rounded-xl m-2 "
                  type="button"
                >
                  cancel
                </button>
              </div>
              <div>
                <button
                  class="hidden text-white px-10 ml-8 mt-16 bg-blue-500 hover:bg-blue-600 text-white font-bold py-2 px-4 rounded-xl m-2 "
                  type="submit"
                >
                  Save
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-white rounded  w-full shadow-md min-h-[70vh] mt-6 pb-10"
    >
      <div>
        <div
          class="flex flex-col tex-center md:flex-row lg:flex-row xl:flex-row items-center justify-center bg-gray-100 mb-10 h-80 md:h-60 lg:h-60 xl:h-60 pb-10 font-raleway"
        >
          <span
            class="w-20 h-20 my-8 md:w-24 lg:w-24 xl:w-24 md:h-24 lg:h-24 xl:h-24 mr-6"
          >
            <div
              class="relative"
            >
              <label
                for=""
              >
                <img
                  alt="profile-img"
                  class=" opacity-100 rounded-full object-cover w-20 h-20 md:w-24 lg:w-24 xl:w-24 md:h-24 lg:h-24 xl:h-24"
                  id="profile-img"
                />
                <div
                  class="hidden absolute bg-gray-100 rounded-full w-5 h-5 left-7 md:left-10 lg:left-10 xl:left-20 top-7 md:top-8 lg:top-16 xl:top-18 cursor-pointer"
                >
                  <img
                    alt=""
                    class="w-3 h-3 m-1"
                    src="camera-solid.svg"
                  />
                </div>
              </label>
            </div>
            <input
              accept=".jpg, .jpeg, .png"
              id="photo-upload"
              type="file"
            />
          </span>
          <div
            class="h-20 flex flex-col items-center md:items-baseline lg:items-baseline xl:items-baseline"
          >
            <h1
              class="font-black text-lg md:text-2xl lg:text-3xl xl:text-3xl "
            >
               
            </h1>
            <h2
              class="font-bold text-sm md:text-base lg:text-base xl:text-base "
            >
              <b
                class="capitalize text-primary"
              />
               
               
            </h2>
            <h3 />
            <div
              class=""
            >
              <div>
                <button
                  class="flex text-white text-sm md:text-base lg:text-base xl:text-base mt-4 px-4 md:px-6 lg:px-6 xl:px-6 bg-blue-500 hover:bg-blue-600 text-white font-bold py-2 px-4 rounded-xl m-2 "
                >
                  Edit profile
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col items-center"
      >
        <form
          action=""
        >
          <section
            class="flex flex-col items-center lg:flex-row xl:flex-row"
          >
            <div
              class="ml-8"
            >
              <div
                class="mb-4 block font-raleway"
              >
                <div
                  class="undefined mb-2 block font-raleway"
                >
                  <label
                    class="block text-grey-darker text-sm font-bold font-rale mb-2"
                    for="FirstName"
                  >
                    First Name
                  </label>
                  <input
                    class="font-rale border-none rounded w-full py-2 px-3 text-grey-darker outline-hidden bg-white text-md appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                    disabled=""
                    id="firstName"
                    name="firstName"
                    value=""
                  />
                </div>
              </div>
              <div
                class="mb-4 block font-raleway"
              >
                <div
                  class="undefined mb-2 block font-raleway"
                >
                  <label
                    class="block text-grey-darker text-sm font-bold font-rale mb-2"
                    for="LastName"
                  >
                    Last name 
                  </label>
                  <input
                    class="font-rale border-none rounded w-full py-2 px-3 text-grey-darker outline-hidden bg-white text-md appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                    disabled=""
                    id="lastName"
                    name="lastName"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="pl-10 lg:pl-24 xl:pl-24"
            >
              <div
                class="mb-4 block font-raleway"
              >
                <div
                  class="undefined mb-2 block font-raleway"
                >
                  <label
                    class="block text-grey-darker text-sm font-bold font-rale mb-2"
                    for="mobileNumber"
                  >
                    Mobile Number 
                  </label>
                  <input
                    class="font-rale border-none rounded w-full py-2 px-3 text-grey-darker outline-hidden bg-white text-md appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                    disabled=""
                    id="mobileNumber"
                    name="mobileNumber"
                    pattern="^07[0-9]{8}$"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="mb-4 block font-raleway"
              >
                <div
                  class="undefined mb-2 block font-raleway"
                >
                  <label
                    class="block text-grey-darker text-sm font-bold font-rale mb-2"
                    for="address"
                  >
                    Address
                  </label>
                  <input
                    class="font-rale border-none rounded w-full py-2 px-3 text-grey-darker outline-hidden bg-white text-md appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                    disabled=""
                    id="address"
                    name="address"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </section>
          <div
            class="flex"
          >
            <div>
              <button
                class="hidden bg-red text-white px-10 ml-8 mt-16 bg-blue-500 hover:bg-blue-600 text-white font-bold py-2 px-4 rounded-xl m-2 "
                type="button"
              >
                cancel
              </button>
            </div>
            <div>
              <button
                class="hidden text-white px-10 ml-8 mt-16 bg-blue-500 hover:bg-blue-600 text-white font-bold py-2 px-4 rounded-xl m-2 "
                type="submit"
              >
                Save
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Profile should render the Profile component 1`] = `
<div
  className="bg-white rounded  w-full shadow-md min-h-[70vh] mt-6 pb-10"
>
  <div>
    <div
      className="flex flex-col tex-center md:flex-row lg:flex-row xl:flex-row items-center justify-center bg-gray-100 mb-10 h-80 md:h-60 lg:h-60 xl:h-60 pb-10 font-raleway"
    >
      <span
        className="w-20 h-20 my-8 md:w-24 lg:w-24 xl:w-24 md:h-24 lg:h-24 xl:h-24 mr-6"
      >
        <div
          className="relative"
        >
          <label
            htmlFor=""
          >
            <img
              alt="profile-img"
              className=" opacity-100 rounded-full object-cover w-20 h-20 md:w-24 lg:w-24 xl:w-24 md:h-24 lg:h-24 xl:h-24"
              id="profile-img"
            />
            <div
              className="hidden absolute bg-gray-100 rounded-full w-5 h-5 left-7 md:left-10 lg:left-10 xl:left-20 top-7 md:top-8 lg:top-16 xl:top-18 cursor-pointer"
            >
              <img
                alt=""
                className="w-3 h-3 m-1"
                src="camera-solid.svg"
              />
            </div>
          </label>
        </div>
        <input
          accept=".jpg, .jpeg, .png"
          id="photo-upload"
          onChange={[Function]}
          type="file"
        />
      </span>
      <div
        className="h-20 flex flex-col items-center md:items-baseline lg:items-baseline xl:items-baseline"
      >
        <h1
          className="font-black text-lg md:text-2xl lg:text-3xl xl:text-3xl "
        >
           
        </h1>
        <h2
          className="font-bold text-sm md:text-base lg:text-base xl:text-base "
        >
          <b
            className="capitalize text-primary"
          />
           
           
        </h2>
        <h3 />
        <div
          className=""
        >
          <div>
            <button
              className="flex text-white text-sm md:text-base lg:text-base xl:text-base mt-4 px-4 md:px-6 lg:px-6 xl:px-6 bg-blue-500 hover:bg-blue-600 text-white font-bold py-2 px-4 rounded-xl m-2 "
              onClick={[Function]}
            >
              Edit profile
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="flex flex-col items-center"
  >
    <form
      action=""
      onSubmit={[Function]}
    >
      <section
        className="flex flex-col items-center lg:flex-row xl:flex-row"
      >
        <div
          className="ml-8"
        >
          <div
            className="mb-4 block font-raleway"
          >
            <div
              className="undefined mb-2 block font-raleway"
            >
              <label
                className="block text-grey-darker text-sm font-bold font-rale mb-2"
                htmlFor="FirstName"
              >
                First Name
              </label>
              <input
                className="font-rale border-none rounded w-full py-2 px-3 text-grey-darker outline-hidden bg-white text-md appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                disabled={true}
                id="firstName"
                name="firstName"
                required={false}
              />
            </div>
          </div>
          <div
            className="mb-4 block font-raleway"
          >
            <div
              className="undefined mb-2 block font-raleway"
            >
              <label
                className="block text-grey-darker text-sm font-bold font-rale mb-2"
                htmlFor="LastName"
              >
                Last name 
              </label>
              <input
                className="font-rale border-none rounded w-full py-2 px-3 text-grey-darker outline-hidden bg-white text-md appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                disabled={true}
                id="lastName"
                name="lastName"
                required={false}
                type="text"
              />
            </div>
          </div>
        </div>
        <div
          className="pl-10 lg:pl-24 xl:pl-24"
        >
          <div
            className="mb-4 block font-raleway"
          >
            <div
              className="undefined mb-2 block font-raleway"
            >
              <label
                className="block text-grey-darker text-sm font-bold font-rale mb-2"
                htmlFor="mobileNumber"
              >
                Mobile Number 
              </label>
              <input
                className="font-rale border-none rounded w-full py-2 px-3 text-grey-darker outline-hidden bg-white text-md appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                disabled={true}
                id="mobileNumber"
                name="mobileNumber"
                pattern="^07[0-9]{8}$"
                required={true}
                type="text"
              />
            </div>
          </div>
          <div
            className="mb-4 block font-raleway"
          >
            <div
              className="undefined mb-2 block font-raleway"
            >
              <label
                className="block text-grey-darker text-sm font-bold font-rale mb-2"
                htmlFor="address"
              >
                Address
              </label>
              <input
                className="font-rale border-none rounded w-full py-2 px-3 text-grey-darker outline-hidden bg-white text-md appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                disabled={true}
                id="address"
                name="address"
                required={false}
                type="text"
              />
            </div>
          </div>
        </div>
      </section>
      <div
        className="flex"
      >
        <div>
          <button
            className="hidden bg-red text-white px-10 ml-8 mt-16 bg-blue-500 hover:bg-blue-600 text-white font-bold py-2 px-4 rounded-xl m-2 "
            type="button"
          >
            cancel
          </button>
        </div>
        <div>
          <button
            className="hidden text-white px-10 ml-8 mt-16 bg-blue-500 hover:bg-blue-600 text-white font-bold py-2 px-4 rounded-xl m-2 "
            type="submit"
          >
            Save
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
